### PR TITLE
#24: Add autoInferenceColumnNames property

### LINKED ISSUE #24: Allow configuring column name conversion for schema auto inference
To allow configuration of how column names are converted to Exasol column names during automatic schema inference (see https://github.com/exasol/virtual-schema-common-document/issues/189) we need to add an option to the EDML schema.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #24

### CODE REVIEW

File: src/main/java/com/exasol/adapter/document/edml/deserializer/EdmlDeserializer.java [hunk 1]
```diff
@@ -38,8 +39,18 @@ public EdmlDefinition deserialize(final String edmlDefinitionAsJson) {
             builder.additionalConfiguration(jsonObjectToString(additionalConfiguration));
         }
         final JsonObject mapping = json.getJsonObject(KEY_MAPPING);
+        final String columnNameMapping = json.getString(AUTO_INFERENCE_COLUMN_NAMES, null);
```
Comment (tomuben) [inline src/main/java/com/exasol/adapter/document/edml/deserializer/EdmlDeserializer.java]:
```suggestion
        final String autoInferenceColumnNames = json.getString(AUTO_INFERENCE_COLUMN_NAMES, null);
```

File: src/main/java/com/exasol/adapter/document/edml/deserializer/EdmlDeserializer.java [hunk 2]
```diff
@@ -38,8 +39,18 @@ public EdmlDefinition deserialize(final String edmlDefinitionAsJson) {
...
             builder.mapping(new MappingDeserializer().deserializeMapping(mapping));
+            if (columnNameMapping != null) {
+                throw new IllegalArgumentException(ExaError.messageBuilder("E-EDML-105")
+                        .message("Auto inference column names is not allowed when mapping is present.")
+                        .mitigation("Remove property {{auto inference property}}.", AUTO_INFERENCE_COLUMN_NAMES)
+                        .toString());
+            }
+        }
+        if (columnNameMapping != null) {
+            builder.autoInferenceColumnNames(ColumnNameMapping.valueOf(columnNameMapping));
```
Comment (tomuben) [inline src/main/java/com/exasol/adapter/document/edml/deserializer/EdmlDeserializer.java]:
```suggestion
            if (autoInferenceColumnNames != null) {
                throw new IllegalArgumentException(ExaError.messageBuilder("E-EDML-105")
                        .message("Parameter auto inference column names is not allowed when mapping is present.")
                        .mitigation("Remove property {{auto inference property}}.", AUTO_INFERENCE_COLUMN_NAMES)
                        .toString());
            }
        }
        if (autoInferenceColumnNames != null) {
            builder.autoInferenceColumnNames(ColumnNameMapping.valueOf(autoInferenceColumnNames));
```

**Review by tomuben — DISMISSED** [review]

File: .gitattributes [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
-.github/workflows/ci-build.yml                                   linguist-generated=true
-.github/workflows/dependencies_check.yml                         linguist-generated=true
-.github/workflows/release_droid_prepare_original_checksum.yml    linguist-generated=true
-.github/workflows/release_droid_print_quick_checksum.yml         linguist-generated=true
-.github/workflows/release_droid_upload_github_release_assets.yml linguist-generated=true
-.github/workflows/release_droid_release_on_maven_central.yml     linguist-generated=true
-
-.settings/org.eclipse.jdt.core.prefs linguist-generated=true
-.settings/org.eclipse.jdt.ui.prefs   linguist-generated=true
+pk_generated_parent.pom                        linguist-generated=true
```
Comment (ckunki) [inline .gitattributes]: Do we want to stick to `linguist-generated=true` or do we prefer the shorter `linguist-generated` ?

File: doc/changes/changes_2.1.0.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# EDML Java 2.1.0, released 2024-06-13
+
+Code name: Add property `autoInferenceColumnNames`
+
+## Summary
+
+This release adds property `autoInferenceColumnNames` to the EDML definition. This allows users to configure how virtual schemas convert column names from the source to Exasol column names when using automatic schema inference.
```
Comment (ckunki) [inline doc/changes/changes_2.1.0.md]: Please add a reference or copy the description about the supported values for this property.
Please specify where to find the property (maybe with a link or reference?)
See file edml-2.1.0.json: 

```json
  "CONVERT_TO_UPPER_SNAKE_CASE",
                "KEEP_SOURCE"
```

File: src/main/java/com/exasol/adapter/document/edml/ColumnNameMapping.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ */
+public enum ColumnNameMapping {
+    /**
+     * Convert column names to {@code UPPER_SNAKE_CASE} (default).
+     */
+    CONVERT_TO_UPPER_SNAKE_CASE,
+    /**
+     * Do not convert column names, use column name from source.
+     */
+    KEEP_SOURCE;
```
Comment (ckunki) [inline src/main/java/com/exasol/adapter/document/edml/ColumnNameMapping.java]:
```suggestion
    KEEP_ORIGINAL_NAME;
```

File: src/main/java/com/exasol/adapter/document/edml/deserializer/EdmlDeserializer.java (hunk 2 above)
Comment (ckunki) [reply src/main/java/com/exasol/adapter/document/edml/deserializer/EdmlDeserializer.java]: I suggest "Property"

**Review by ckunki — CHANGES_REQUESTED** [review]

**Review by ckunki — APPROVED** [review]

**Pull request merged**